From ea2d9be01d383bc880adc4f6dc4f6fa601a6d50f Mon Sep 17 00:00:00 2001
From: goutongchen <goutongchen@uniontech.com>
Date: Fri, 29 Jul 2022 11:05:52 +0800
Subject: [PATCH] Fix kabi header file being repeatedly referenced by the same
 file
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: ea2d9be01d383bc880adc4f6dc4f6fa601a6d50f
Modified-by-SEL: No


uniontech inclusion
category: cleanup
bugzilla: https://gitee.com/openeuler/kernel/issues/I5JI5Z
CVE: NA

---------------------------------------

In the following header files, the "linux/kabi.h" header file is
repeatedly referenced twice, causing the same header file to be
included repeatedly:

	include/linux/blk_types.h
	include/linux/cred.h
	include/linux/device.h
	include/linux/device/class.h
	include/linux/ioport.h
	include/linux/mm.h
	include/linux/swap.h

Remove redundant "linux/kabi.h" header file references to make
the code more canonical.

Signed-off-by: goutongchen <goutongchen@uniontech.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/blk_types.h    | 1 -
 include/linux/cred.h         | 1 -
 include/linux/device.h       | 1 -
 include/linux/device/class.h | 1 -
 include/linux/ioport.h       | 1 -
 include/linux/mm.h           | 1 -
 include/linux/swap.h         | 1 -
 7 files changed, 7 deletions(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index 11b9505b14c6..a0ec5ff216ff 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -6,7 +6,6 @@
 #ifndef __LINUX_BLK_TYPES_H
 #define __LINUX_BLK_TYPES_H
 
-#include <linux/kabi.h>
 #include <linux/types.h>
 #include <linux/bvec.h>
 #include <linux/ktime.h>
diff --git a/include/linux/cred.h b/include/linux/cred.h
index cd1b5fc47d52..7385295198ca 100644
--- a/include/linux/cred.h
+++ b/include/linux/cred.h
@@ -8,7 +8,6 @@
 #ifndef _LINUX_CRED_H
 #define _LINUX_CRED_H
 
-#include <linux/kabi.h>
 #include <linux/capability.h>
 #include <linux/init.h>
 #include <linux/key.h>
diff --git a/include/linux/device.h b/include/linux/device.h
index 929b66dc5ea9..d9291e2a624a 100644
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -12,7 +12,6 @@
 #ifndef _DEVICE_H_
 #define _DEVICE_H_
 
-#include <linux/kabi.h>
 #include <linux/dev_printk.h>
 #include <linux/energy_model.h>
 #include <linux/ioport.h>
diff --git a/include/linux/device/class.h b/include/linux/device/class.h
index f2071ee10d61..34f259326731 100644
--- a/include/linux/device/class.h
+++ b/include/linux/device/class.h
@@ -14,7 +14,6 @@
 #ifndef _DEVICE_CLASS_H_
 #define _DEVICE_CLASS_H_
 
-#include <linux/kabi.h>
 #include <linux/kobject.h>
 #include <linux/klist.h>
 #include <linux/pm.h>
diff --git a/include/linux/ioport.h b/include/linux/ioport.h
index fc7e99ba6420..3930f564f207 100644
--- a/include/linux/ioport.h
+++ b/include/linux/ioport.h
@@ -10,7 +10,6 @@
 #define _LINUX_IOPORT_H
 
 #ifndef __ASSEMBLY__
-#include <linux/kabi.h>
 #include <linux/compiler.h>
 #include <linux/types.h>
 #include <linux/bits.h>
diff --git a/include/linux/mm.h b/include/linux/mm.h
index a886f48b6a0e..dd15c6ed8ae7 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -6,7 +6,6 @@
 
 #ifdef __KERNEL__
 
-#include <linux/kabi.h>
 #include <linux/mmdebug.h>
 #include <linux/gfp.h>
 #include <linux/bug.h>
diff --git a/include/linux/swap.h b/include/linux/swap.h
index 2b68047db2d9..f2aa72ec0e57 100644
--- a/include/linux/swap.h
+++ b/include/linux/swap.h
@@ -2,7 +2,6 @@
 #ifndef _LINUX_SWAP_H
 #define _LINUX_SWAP_H
 
-#include <linux/kabi.h>
 #include <linux/spinlock.h>
 #include <linux/linkage.h>
 #include <linux/mmzone.h>
-- 
2.34.1

